src/Razor/src/Microsoft.VisualStudio.RazorExtension/RazorPackage.cs

```diff
@@ -81,8 +87,6 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             mcs.AddCommand(menuToolWin);
         }
 
-        var componentModel = (IComponentModel)GetGlobalService(typeof(SComponentModel));
```

Comment:
To my reading of this change, it seems the code from line 65 to 89 is somehow transitioning us to a background thread? That seems surprising. Do you know which thing does it? If so, perhaps a comment.

Reply:
I expect that it's caused by the `await` on 76. I find that surprising as well. Would be better to leave the code here and ensure that we switch to the main thread rather than relying on the threading behavior across the method body? Something like this?

```C#
if (await GetServiceAsync(typeof(SComponentModel)) is IComponentModel componentModel)
{
    // Comment mentioning that we need to ensure that OptionsStorage needs to be created on the main thread.
    await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);

    _optionsStorage = componentModel.GetService<OptionsStorage>();
}
```

Reply:
This one is a bit tough to explain succinctly, but basically anything involved in using `IRazorLoggerFactory` and creating a logger that is not on the main thread means that it can happen before line 85. 

In the case of the original bug filing, it was 

```
microsoft.codeanalysis.externalaccess.razor.ni!RazorDynamicFileInfoProviderWrapper.EnsureAttached
```

Reply:
In that case, would it be better to force create the razor logger factory earlier on the main thread?